wallet/src/actors/worker/methods.rs

```diff
@@ -223,7 +241,12 @@ impl Worker {
         block: &model::BlockInfo,
         txns: &[types::Transaction],
     ) -> Result<()> {
+        log::debug!("trying to index txns from epoch {}", block.epoch);
         wallet.index_transactions(block, txns)?;
+        wallet.update_last_sync(CheckpointBeacon {
+            checkpoint: block.epoch,
+            hash_prev_block: Hash::from(block.hash.clone()),
```

Comment:
Maybe instead of implementing conversions between `Vec<u8>` and `Hash`, it would be better to just change the type of `block_info.hash_prev_block` to `Hash`.

```suggestion
            hash_prev_block: block.hash.clone(),
```

Reply:
Great suggestion. I'll measure the impact and see if the moment is now.

Reply:
Done. Thanks once again!